config/src/test/java/org/springframework/security/config/annotation/web/configuration/HttpSecurityConfigurationTests.java

```diff
@@ -374,6 +374,15 @@ public void configureWhenCorsConfigurationSourceThenApplyCors() {
 		assertThat(configSource).isInstanceOf(UrlBasedCorsConfigurationSource.class);
 	}
 
+	@Test
+	public void configureWhenNoUrlBasedCorsConfigThenNoCorsApplied() throws Exception {
+		this.spring.register(NonUrlBasedCorsConfig.class, DefaultWithFilterChainConfig.class).autowire();
+		SecurityFilterChain filterChain = this.spring.getContext().getBean(SecurityFilterChain.class);
+		assertThat(filterChain.getFilters()).noneMatch((filter) -> filter instanceof CorsFilter);
+
+		this.mockMvc.perform(formLogin()).andExpect(header().doesNotExist("Access-Control-Allow-Origin"));
+	}
+
```

Comment:
Can you please add a test that verifies if the header `Vary` is not present? In summary, simulate the problem reported in https://github.com/spring-projects/spring-security/issues/15378 and assert that it is fixed.

You can add the issue number in the test, like so:

```java
// gh-15378
@Test
void ...() {
}
```

Reply:
I have updated the existing test code to verify if the Vary header is not present. Thank you for your feedback!